src/components/Buy/InventoryTable.jsx

```diff
@@ -0,0 +1,91 @@
+import { useEffect } from 'react';
+import { collection, getDocs, getDoc } from 'firebase/firestore/lite';
+import { MDBDataTable } from 'mdbreact';
+import { db } from '../../firebase/firebase_config';
+
+import { buyColumn, noBuyColumns } from './column';
+import './InventoryTable.css';
+
+function InventoryTable({ buyFunctionality, tableData, setTableData, handleAddToCart }) {
+  // This function builds the data we want to display in the inventory
+  // it builds this through book the current textbook information and getting
+  // information about the seller (for the payment methods accepted)
+
+  async function tableFormatBook(book) {
+    const bookData = book.data();
+    const { seller } = bookData;
+    let maybeSellerPaymentMethods = '';
+    // Get the payment methods info
+    try {
+      const maybeSeller = await getDoc(seller);
+      const maybeSellerData = maybeSeller.data();
```

Comment:
instead of these "maybe" variables, i would suggest throwing an error if these do not exist